src/query.c

```diff
@@ -26,6 +26,7 @@
 #include "module.h"
 
 #define EFFECTIVE_FIELDMASK(q_, qn_) ((qn_)->opts.fieldMask & (q)->opts->fieldmask)
+#define UNION_QUICK_EXIT q->opts->quickExit
```

Comment:
Maybe like this?
#define UNION_QUICK_EXIT(q) q->opts->quickExit
So you will not have to count on having a varaible called q?